bigcache_test.go

```diff
@@ -113,6 +115,36 @@ func TestEntryUpdate(t *testing.T) {
 	assert.Equal(t, []byte("value2"), cachedValue)
 }
 
+func TestEntriesMatchingWithoutFilter(t *testing.T) {
+	//t.Parallel()
+
+	// given
+	keysCount := 10000
+	cache, _ := NewBigCache(Config{8, 6 * time.Second, 1, 256, false, nil, 0, nil})
+	value := []byte("value")
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	for i := 0; i < keysCount; i++ {
+		cache.Set(fmt.Sprintf("key%d", i), value)
+	}
+
+	// when
+	keys := make(map[string]struct{})
+	iterator := cache.Iterator()
+
+	for iterator.Next() {
+		current, err := iterator.Value()
+
+		if err == nil {
```

Comment:
Add test when error occures: modify map when it terates, delete element under `Value()`